lnwallet/chancloser/chancloser.go

```diff
@@ -162,23 +231,16 @@ func NewChanCloser(cfg ChanCloseCfg, deliveryScript []byte,
 	idealFeePerKw chainfee.SatPerKWeight, negotiationHeight uint32,
 	closeReq *htlcswitch.ChanClose, locallyInitiated bool) *ChanCloser {
 
-	// Given the target fee-per-kw, we'll compute what our ideal _total_ fee
-	// will be starting at for this fee negotiation.
-	//
-	// TODO(roasbeef): should factor in minimal commit
+	// Given the target fee-per-kw, we'll compute what our ideal _total_
+	// fee will be starting at for this fee negotiation.
 	idealFeeSat := cfg.Channel.CalcFee(idealFeePerKw)
 
-	// If this fee is greater than the fee currently present within the
-	// commitment transaction, then we'll clamp it down to be within the proper
-	// range.
-	//
-	// TODO(roasbeef): clamp fee func?
-	channelCommitFee := cfg.Channel.StateSnapshot().CommitFee
-	if idealFeeSat > channelCommitFee {
-		chancloserLog.Infof("Ideal starting fee of %v is greater than commit "+
-			"fee of %v, clamping", int64(idealFeeSat), int64(channelCommitFee))
-
-		idealFeeSat = channelCommitFee
+	// When we're the initiator, we'll want to also factor in the highest
+	// fee we want to pay. This'll either be 3x the ideal fee, or the
+	// specified explicit max fee.
+	maxFee := idealFeeSat * defaultMaxFeeMultiplier
+	if cfg.MaxFee > 0 {
+		maxFee = cfg.Channel.CalcFee(cfg.MaxFee)
```

Comment:
Similar to the ideal fee, the max fee isn't respected on restart

Reply:
Hmm, I think one way to resolve this would be to commit this information to disk when we call `MarkCoopBroadcasted`.

Reply:
So we'd store persistently the preferences of the user at that point, and for the case where we're the responder (but still the initiator), we'd do something like base it off some (larger?) multiplier of the max commitment fee.

Reply:
I don't think it should be stored in `MarkCoopBroadcasted` since https://github.com/lightningnetwork/lnd/pull/6760 will end up changing that function's usage. This is because the link may still need to be loaded, but we still want the user's preferences. I made a function called `PersistDeliveryScript` which could be extended to persist other information before calling `MarkCoopBroadcasted`

I also think we could check here that the ideal fee isn't greater than the max fee if one exists

Reply:
So from my reading, we'll retain the "ideal" fee on restart, since we'll compute the "ideal" fee based on `CoopCloseTargetConfs` (defaults to 6 blocks). At this point, the `req` will be nil, so we'll then based the max fee off 3x this ideal fee. 

So I think we can either leave it as is, persist it in #6760, or add a config option (yet another one, that ppl prob won't set tbh) to let ppl express what this max multiplier should be.

Reply:
There are two ways the ideal fee is set:
- through rpc if initiating the close
- via EstimateFeePerKW on CoopCloseTargetConfs

The first case probably won't have the same fee on restart. The second may if the fee estimation doesn't change. I don't think this should be persisted in this PR. It could be done in #6760. But I think any persistence changes should be done in the same release so we don't have to check whether something is stored or not

Reply:
SGTM. I'd say a user's _expectation_ would be for things to resume with their desired fee, so we should strive for that. Agree we can do it in another PR, though rn https://github.com/lightningnetwork/lnd/pull/6760 is marked for 0.16 (and in a draft state).